app/models/notification.rb

```diff
@@ -0,0 +1,55 @@
+class Notification < ActiveRecord::Base
```

Comment:
Your methods do not show proper levels of abstraction. It is confusing for me to read your code. Please refactor.

The views should be placed locally within the component. If I'm doing announcements I should be placing the notification views in the announcements view directory.

Existing notifications refer to the notifications that v1 currently sends. How do you map v1 notifications to your framework? Have you considered that?